cpp/src/arrow/sparse_tensor.h

```diff
@@ -120,6 +121,17 @@ class ARROW_EXPORT SparseCOOIndex : public internal::SparseIndexBase<SparseCOOIn
     return indices()->Equals(*other.indices());
   }
 
+  Status ValidateShape(const std::vector<int64_t>& shape) const override {
```

Comment:
Is it useful to inline the implementation here?

Reply:
Yes it is.  I'll make it inline.